bus-mapping/src/evm/stack.rs

```diff
@@ -85,8 +80,85 @@ impl Stack {
         StackAddress::from(1024 - self.0.len())
     }
 
+    /// Returns the second last filled `StackAddress`.
+    pub fn nth_last_filled(&self, nth: usize) -> StackAddress {
+        StackAddress::from(1024 - self.0.len() + nth)
+    }
+
     /// Returns the last [`EvmWord`] allocated in the `Stack`.
     pub fn last(&self) -> Option<&EvmWord> {
         self.0.last()
     }
+
+    /// Returns the second last [`EvmWord`] allocated in the `Stack`.
+    pub fn nth_last(&self, nth: usize) -> Option<&EvmWord> {
+        self.0.get(self.0.len() - (nth + 1))
+    }
+
+    /// Add last and second last value, and store on second last address and delete last element in the `Stack`.
+    pub fn add(&mut self) -> Option<EvmWord> {
+        let length = self.0.len();
+        let last_value = self.0.last().unwrap();
+        let second_last_value = self.0.get(self.0.len() - 2).unwrap();
+        let sum = last_value.adc(*second_last_value).unwrap();
+        self.0[length - 2] = sum;
+        self.pop()
+    }
+
+    /// Pop the last [`EvmWord`] allocated in the `Stack`.
+    pub fn pop(&mut self) -> Option<EvmWord> {
+        self.0.pop()
+    }
+}
+
+#[cfg(test)]
+mod stack_tests {
+    use super::*;
+
+    macro_rules! test_vec {
+        ($($x:expr),*) => (Stack::new(vec![ $(EvmWord::from_str($x).unwrap()), *]))
+    }
```

Comment:
I would avoid macro usage If possible. 

Macros should only be used when there's no other alternative or they simplify a lot the code. In this case, they just remove a bit of verbosity in a single file. And the same thing can be achieved using functions.

Any thoughts @ed255 ?

Reply:
Agree!  Although I think in the future it would be good to have a macro for testing to create EvmWords / Addresses from a string to avoid verbosity, specially in the circuits crate. Something like `word!("0x1003e2d2584a6f10d88b")`.